src/olympia/users/tasks.py

```diff
@@ -65,3 +77,49 @@ def update_user_ratings_task(data, **kw):
     )
     for pk, rating in data:
         UserProfile.objects.filter(pk=pk).update(averagerating=round(float(rating), 2))
+
+
+BATCH_SIZE = 100
+
+
+@task(autoretry_for=(HTTPError, Timeout), max_retries=5, retry_backoff=True)
+def sync_blocked_emails(batch_size=100, **kw):
+    url = f"""
+        {settings.SOCKET_LABS_HOST}/servers/{settings.SOCKET_LABS_SERVER_ID}/suppressions/download?sortField=suppressionLastUpdate&sortDirection=dsc
+    """.strip()
+    headers = {
+        'authorization': f'Bearer {settings.SOCKET_LABS_TOKEN}',
+    }
+    response = requests.get(url, headers=headers)
+
+    # Raise exception if not 200 like response
+    response.raise_for_status()
+
+    # Convert bytes to a file-like object
+    file_like_object = io.BytesIO(response.content)
+    # Create a Django ContentFile from the file-like object
+    django_file = ContentFile(file_like_object.read())
+    # Get 10 character uinique ID for file
+    file_id = str(uuid.uuid4())[:10]
+
+    # save csv to disk
+    file_path = default_storage.save(f'tmp/suppressions-{file_id}.csv', django_file)
+
+    task_log.info(f'Downloaded suppression list to {file_path}')
+
+    with default_storage.open(file_path, 'r') as csv_file:
```

Comment:
if you use tempfile with the `with` it'll clean-up without having to manually `delete` at the end too.

Reply:
Nice, very cool.